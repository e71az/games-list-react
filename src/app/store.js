import { configureStore } from '@reduxjs/toolkit';
import recipeReducer from '../reducers/recipeSlice';

const store = configureStore({
  reducer: {
    recipe: recipeReducer,
  },
});

export default store;
