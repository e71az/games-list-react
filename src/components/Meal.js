/* eslint-disable react/forbid-prop-types, jsx-a11y/anchor-is-valid */
import { PropTypes } from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { recipeFilter } from '../reducers/recipeSlice';

const Meal = ({ meal: { idMeal, strMeal, strMealThumb } }) => {
  const dispatch = useDispatch();

  return (
    <Link to="/recipe" onClick={() => dispatch(recipeFilter(idMeal))}>
      <div
        data-testid="meal"
        className="img-meal-container"
        style={{
          backgroundImage: `url(${strMealThumb})`,
        }}
      >
        <h5 className="transparent-banner pl-3 py-2">{strMeal}</h5>
      </div>
    </Link>
  );
};

Meal.propTypes = {
  meal: PropTypes.object.isRequired,
};

export default Meal;
