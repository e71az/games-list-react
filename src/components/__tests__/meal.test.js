import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../app/store';
import Meal from '../Meal';

afterEach(() => {
  cleanup();
});

describe('Meal component tests', () => {
  const meal = {
    idMeal: 1,
    strMeal: 'Shrimp Pizza',
    strMealThumb:
      'https://cdn2.cocinadelirante.com/sites/default/files/styles/gallerie/public/images/2017/04/pizzapepperoni0.jpg',
  };

  // This render is for normal testing
  render(
    <Provider store={store}>
      <Router>
        <Meal meal={meal} />
      </Router>
    </Provider>,
  );

  // This renderer is for snapshot testing
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Meal meal={meal} />
        </Router>
      </Provider>,
    )
    .toJSON();

  const mealElement = screen.getByTestId('meal');

  test('Component should Render', () => {
    expect(mealElement).toBeInTheDocument();
  });
  test('Component should contain Shrimp Pizza title', () => {
    expect(mealElement).toHaveTextContent('Shrimp Pizza');
  });
  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
