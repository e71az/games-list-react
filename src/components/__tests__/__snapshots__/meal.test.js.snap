// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal component tests Matches snapshot 1`] = `
<a
  href="/recipe"
  onClick={[Function]}
>
  <div
    className="img-meal-container"
    data-testid="meal"
    style={
      Object {
        "backgroundImage": "url(https://cdn2.cocinadelirante.com/sites/default/files/styles/gallerie/public/images/2017/04/pizzapepperoni0.jpg)",
      }
    }
  >
    <h5
      className="transparent-banner pl-3 py-2"
    >
      Shrimp Pizza
    </h5>
  </div>
</a>
`;
